lib/Messenger/Handler/SearchBackendHandler.php

```diff
@@ -35,6 +38,29 @@ public function __invoke(SearchBackendMessage $message, Acknowledger $ack = null
         return $this->handle($message, $ack);
     }
 
+    private function processElement(Element\ElementInterface $element, bool $updateChildren) {
+
+        $searchEntry = Data::getForElement($element);
+        if ($searchEntry->getId() instanceof Data\Id) {
+            $searchEntry->setDataFromElement($element);
+        } else {
+            $searchEntry = new Data($element);
+        }
+        $searchEntry->save();
+
+        if ($updateChildren) {
+            if ($element instanceof Asset || $element instanceof AbstractObject || $element instanceof Document) {
+                foreach ($element->getChildren() as $child) {
```

Comment:
We can't use `getChildren()` here, because this returns only direct children (1st level). 
I think we should anyway consider to use plain SQL to update all the children for performance reasons, just as we do e.g. here: 
https://github.com/pimcore/pimcore/blob/70b1db360a2a4fa498c69f71a33aa1db5a3a42d1/models/Asset/Dao.php#L213 

This should speed up things a lot and make it more scalable (when it comes to millions of children).